java/java.editor/src/org/netbeans/modules/editor/java/TypingCompletion.java

```diff
@@ -547,6 +547,11 @@ static boolean posWithinString(Document doc, int caretOffset) {
         return posWithinQuotes(doc, caretOffset, JavaTokenId.STRING_LITERAL);
     }
     
+ static boolean posWithinTextBlock(Document doc, int caretOffset) {
+        TokenSequence<JavaTokenId> javaTS=javaTokenSequence(doc,caretOffset, false);
+        javaTS.movePrevious();
```

Comment:
check if `javaTS` is not null and that `javaTS.movePrevious` actually didn't return `false` (there would be no token for `javaTS.token()`